### ISSUE TITLE
'Use strict' in pdfjs context only, compatibility.js

### ISSUE THREAD

**Issue by McGiogen** [mozilla/pdf.js#7307]
Hi,
We have a problem using compatibility.js caused by the absence of a local context to the file and by the use of 'use strict' directive.
The problem is really simple: the file starts with 'use strict' without creating a local context. In our application we minify all the external dependencies in a single file and because of compatibility.js every file minified after it is in its 'use strict' context.
This is not the correct behaviour (and, in our case, fires blocking exceptions) and has been already solved in the other js files of this repository (pdf.js, pdf.worker.js, pdf.combined.js, pdf_viewer.js) which starts with the following lines:
```
(function pdfjsWrapper() {
  // Use strict in our context only - users might not want it
  'use strict';
```
compatibility.js instead starts simply with:
```
'use strict';
```

So also compatibility.js should have a local context. I can do a pr if it helps.
Thanks,
McGio

**Comment by MooDySmiles** [comment]
+1

**Comment by alessandroneri90** [comment]
+1

**Comment by Roqk** [comment]
+1

**Comment by Snuffleupagus** [comment]
Please ***do not*** post "+1" comments (and similar), since it generates *a lot* of unnecessary notifications for people. Instead, please use the GitHub reaction feature, see https://help.github.com/articles/about-discussions-in-issues-and-pull-requests/#reacting-to-ideas-in-issues-and-pull-requests and https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments.

**Comment by yurydelendik** [comment]
I think it was the same person who "+1" (just from multiple account), so we can safely deprioritize the request.

**Comment by yurydelendik** [comment]
> So also compatibility.js should have a local context. I can do a pr if it helps.

@McGiogen yes please. There is no need in space/tab indenting of entire compatibility.js. You may need to modify jshint settings. See https://github.com/mozilla/pdf.js/wiki/Contributing prior submitting a PR.

**Comment by yurydelendik** [comment]
Closed by #7315
